File: treeView/src/main/java/org/sqlunet/treeview/control/RootController.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.treeview.control;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import org.sqlunet.treeview.model.TreeNode;
import org.sqlunet.treeview.view.SubtreeView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class RootController extends Controller<Void>
{
	@Nullable
	private ViewGroup contentView;

	/**
	 * Constructor
	 */
	public RootController()
	{
		super(false);
	}

	@Nullable
	@Override
	public View createNodeView(@NonNull final Context context, final TreeNode node, final Void value)
	{
		return null;
	}

	@Nullable
	@Override
	public ViewGroup getChildrenView()
	{
		return this.contentView;
	}

	@Nullable
	@Override
	public SubtreeView getSubtreeView()
	{
		if (this.contentView != null && this.contentView.getChildCount() > 0)
		{
			return (SubtreeView) this.contentView.getChildAt(0);
		}
		return null;
	}

	public void setContentView(@Nullable final ViewGroup contentView)
	{
		this.contentView = contentView;
	}
}

File: treeView/src/main/java/org/sqlunet/treeview/control/RootController.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.treeview.control

import android.content.Context
import android.view.View
import android.view.ViewGroup
import org.sqlunet.treeview.model.TreeNode
import org.sqlunet.treeview.view.SubtreeView

class RootController : Controller(false) {

    override fun createNodeView(context: Context, node: TreeNode, minHeight: Int): View? {
        return null
    }

    override var childrenView: ViewGroup? = null
        set(childrenView) {
            super.childrenView = childrenView
            field = childrenView
        }

    override var subtreeView: SubtreeView?
        get() = if (childrenView != null && childrenView!!.childCount > 0) {
            childrenView!!.getChildAt(0) as SubtreeView
        } else null
        set(subtreeView) {
            super.subtreeView = subtreeView
        }

    fun setContentView(contentView: ViewGroup?) {
        childrenView = contentView
    }
}
